File: Com.Fubei.Api.OpenApi/Biz/AgentApi/Model/Request/AOrderPayParam.cs
using System;
using System.Collections.Generic;
using Com.Illuminati.Galileo.Model;
using Newtonsoft.Json;

namespace Com.Fubei.Api.OpenApi.Biz.AgentApi.Model.Request
{
    /// <summary>
    /// 付款码支付
    /// 该接口支持服务商级和商户级接入。
    /// http://docs.51fubei.com/agent-api/payment/orderPay.html
    /// </summary>
    public class AOrderPayParam : BaseEntity
    {
        [JsonProperty("merchant_order_sn")]
        public string MerchantOrderSn { get; set; }

        [JsonProperty("merchant_id")]
        public int? MerchantId { get; set; }

        [JsonProperty("auth_code")]
        public string AuthCode { get; set; }

        [JsonProperty("total_amount")]
        public decimal TotalAmount { get; set; }

        [JsonProperty("store_id")]
        public int? StoreId { get; set; }

        [JsonProperty("cashier_id")]
        public int? CashierId { get; set; }

        [JsonProperty("sub_appid")]
        public string SubAppId { get; set; }

        [JsonProperty("goods_tag")]
        public string GoodsTag { get; set; }

        [JsonProperty("detail")]
        public GoodsDetailEntity Detail { get; set; }

        [JsonProperty("device_no")]
        public string DeviceNo { get; set; }

        [JsonProperty("body")]
        public string Body { get; set; }

        [JsonProperty("attach")]
        public string Attach { get; set; }

        [JsonIgnore]
        public string TimeoutExpire
        {
            get => TimeoutExpress;
            set => TimeoutExpress = value;
        }

        [JsonProperty("timeout_express")]
        public string TimeoutExpress { get; set; }
        
        [JsonProperty("notify_url")]
        public string NotifyUrl { get; set; }

        [JsonProperty("alipay_extend_params")]
        public AlipayExtendParamsEntity AlipayExtendParams { get; set; }
    }
}

File: Com.Fubei.OpenApi.Vb/Biz/AgentApi/Model/Request/AOrderPayParam.vb
Imports Com.Illuminati.Galileo.Model
Imports Newtonsoft.Json

Namespace Biz.V2.Model.Request
    
    ' 付款码支付
    ' http://docs.51fubei.com/agent-api/payment/orderPay.html
    Public Class AOrderPayParam
        Inherits BaseEntity

        <JsonProperty("merchant_order_sn")>
        Public Property MerchantOrderSn As String

        <JsonProperty("merchant_id")>
        Public Property MerchantId As Integer?

        <JsonProperty("auth_code")>
        Public Property AuthCode As String

        <JsonProperty("total_amount")>
        Public Property TotalAmount As Decimal

        <JsonProperty("store_id")>
        Public Property StoreId As Integer?

        <JsonProperty("cashier_id")>
        Public Property CashierId As Integer?

        <JsonProperty("sub_appid")>
        Public Property SubAppId As String

        <JsonProperty("goods_tag")>
        Public Property GoodsTag As String

        <JsonProperty("detail")>
        Public Property Detail As GoodsDetailEntity

        <JsonProperty("device_no")>
        Public Property DeviceNo As String
        <JsonProperty("body")>
        Public Property Body As String

        <JsonProperty("attach")>
        Public Property Attach As String

        <JsonProperty("timeout_express")>
        Public Property TimeoutExpress As String

        <JsonProperty("notify_url")>
        Public Property NotifyUrl As String

        <JsonProperty("alipay_extend_params")>
        Public Property AlipayExtendParams As AlipayExtendParamsEntity
    End Class
End Namespace
